src/message_bus.zig

```diff
@@ -212,6 +212,16 @@ fn MessageBusType(comptime process_type: vsr.ProcessType) type {
                 } else null,
                 .user_timeout_ms = constants.tcp_user_timeout_ms,
                 .nodelay = constants.tcp_nodelay,
+            });
+            return fd;
+        }
+
+        fn init_tcp_listen(io: *IO, address: std.net.Address) !struct {
+            fd: IO.socket_t,
+            address: std.net.Address,
+        } {
+            const fd = try init_tcp(io, address.any.family);
```

Comment:
Given that we've extracted init_tcp, we can now pass `fd` as a parameter, and simply the signature here to return just address.

Also, `errdefer close(fd)` is needed.